specification/cognitiveservices/OpenAI.Inference/client.tsp

```diff
@@ -5,15 +5,25 @@ using Azure.ClientGenerator.Core;
 
 // Azure-specific long-running operations should be treated as implementation details that are wrapped into
 // appropriately merged public surface.
-@@internal(Azure.OpenAI.beginAzureBatchImageGeneration);
-@@internal(Azure.OpenAI.getAzureBatchImageGenerationOperationStatus);
+@@access(Azure.OpenAI.beginAzureBatchImageGeneration, Access.internal);
+@@access(Azure.OpenAI.getAzureBatchImageGenerationOperationStatus, Access.internal);
 
 // Azure-specific Chat Completions with extensions should be handled by clients as a conditional selection within the
 // shared Chat Completions route, with the selection gated by the presence or non-presence of additional child
 // configuration options on the request payload options model.
-@@internal(Azure.OpenAI.getChatCompletionsWithAzureExtensions);
+@@access(Azure.OpenAI.getChatCompletionsWithAzureExtensions, Access.internal);
 
 // Some models from routes with suppressed visibility are still desired for custom public surface.
-@@include(Azure.OpenAI.ImageGenerationOptions);
-@@include(Azure.OpenAI.ImageLocation);
-@@include(Azure.OpenAI.ImageGenerations);
+@@access(Azure.OpenAI.ImageGenerationOptions, Access.public);
+@@access(Azure.OpenAI.ImageLocation, Access.public);
+@@access(Azure.OpenAI.ImageGenerations, Access.public);
```

Comment:
`ImageGenerations` has a union type which is not supported.

Reply:
If keeps `@@access(Azure.OpenAI.ImageGenerations, Access.public);` unchange, and adopts all new changes in Java, it works fine.

Reply:
Thanks Shawn. If OpenAI is good with it, I can do one more test with their latest typespec on main, and update the Java list.

---

Java is designed to fail if one specify a model refs Union as `public` (agreed from Srikanta). That is the reason that typespec-java fails when e.g. `@@access(Azure.OpenAI.ImageGenerationOptions, Access.public);`

If other languages need it, currently it had to be done with e.g.
``` ts
@@access(Azure.OpenAI.ImageGenerationOptions, Access.public, "csharp");
@@access(Azure.OpenAI.ImageGenerationOptions, Access.public, "python");
@@access(Azure.OpenAI.ImageGenerationOptions, Access.public, "javascript");
```

Reply:
The client.tsp changes in Spec PR works for Java

Reply:
> Thanks Shawn. If OpenAI is good with the changes, I can do one more test with their latest typespec on main, and update the Java list.
> 
> Java is designed to fail if one specify a model refs Union as `Access.public` (agreed from Srikanta). That is the reason that typespec-java fails when e.g. `@@access(Azure.OpenAI.ImageGenerations, Access.public);`
> 
> If other languages need it, I think currently it had to be done with e.g.
> 
> ```ts
> @@access(Azure.OpenAI.ImageGenerations, Access.public, "csharp");
> @@access(Azure.OpenAI.ImageGenerations, Access.public, "python");
> @@access(Azure.OpenAI.ImageGenerations, Access.public, "javascript");
> ```

Hi @weidongxu-microsoft, do you know there is a way to exclude the `java` in the scope only but not list all other languages?

Reply:
"exclude sdk in scope" is in plan but not implemented in TCGC yet.